tests/e2e-leg-9/replicated-upgrade-sanity/38-assert.yaml

```diff
@@ -0,0 +1,26 @@
+# (c) Copyright [2021-2024] Open Text.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# You may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+apiVersion: v1
+kind: Event
+reason: ReplicationSucceeded
+source:
+  component: verticadb-operator
+involvedObject:
+  apiVersion: vertica.com/v1beta1
+  kind: VerticaReplicator
+---
+apiVersion: vertica.com/v1beta1
+kind: VerticaReplicator
```

Comment:
I think there is a logic in `replicatedupgrade_reconciler` that deletes the `vrep` CR once the replication finishes (either success or failure), how do we make sure the CR still exists at this step? And do we plan to address the operator logic issue in a future PR, i.e. the operator should exit reconciler loop with failure once it detects the replication is unsuccessful or it should have a loop which creates `vrep` CR repetitively until replication is successful. Currently the operator doesn't seem to distinguish replication success or failure (it only detects completion of a replication).

Reply:
You mean as long as the data insertion finishes after sandbox is created successfully and before replication is started by the operator we should be able to pass, right? I'm curious whether we can guarantee this or at least make sure this is the case for most of the time. Looking at the logic of `replicatedupgrade_reconciler` I see that we start draining connections to replica group A right after sandbox is created and before replication is started.
```
// Upgrade the version in the sandbox to the new version.
r.postUpgradeSandboxMsg,
r.upgradeSandbox,
r.waitForSandboxUpgrade,
// Pause all connections to replica A. This is to prepare for the
// replication below.
r.postPauseConnectionsMsg,
r.pauseConnectionsAtReplicaGroupA,
// Copy any new data that was added since the sandbox from replica group
// A to replica group B.
r.postStartReplicationMsg,
r.startReplicationToReplicaGroupB,
r.waitForReplicateToReplicaGroupB,
```
So what we want is that the data insertion happens exactly at this stage, i.e. while we are draining connections and strictly before we finish draining? Then how long does this step of draining connections usually take? We know that data insertion takes about 1 second, right?

Reply:
It's not only draining connections. The data insertion happens right after `r.sandboxReplicaGroupB`. We should be fine as long as data insertion finishes before the below steps:
		r.postUpgradeSandboxMsg,
		r.upgradeSandbox,
		r.waitForSandboxUpgrade,
		// Pause all connections to replica A. This is to prepare for the
		// replication below.
		r.postPauseConnectionsMsg,
		r.pauseConnectionsAtReplicaGroupA,